deltachat-ios/TextFieldCell.swift

```diff
@@ -9,85 +9,113 @@
-    passwordCell.textField.isSecureTextEntry = true
+    private func setupViews() {
+        contentView.addSubview(textField)
+        textField.translatesAutoresizingMaskIntoConstraints = false
+        let margins = contentView.layoutMarginsGuide
+        let trailing = margins.trailingAnchor
+        textField.trailingAnchor.constraint(equalTo: trailing).isActive = true
+        textField.centerYAnchor.constraint(equalTo: contentView.centerYAnchor).isActive = true
+        if let label = self.textLabel {
+            textField.leadingAnchor.constraint(equalTo: label.trailingAnchor, constant: 20).isActive = true // this will prevent the textfield from growing over the textLabel while typing
+        } else {
+            textField.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 20).isActive = true
+        }
+    }
 
-    return passwordCell
-  }
+    override func setSelected(_ selected: Bool, animated _: Bool) {
+        if selected {
+            textField.becomeFirstResponder()
+        }
+    }
 
-  static func makeNameCell() -> TextFieldCell {
-    let nameCell = TextFieldCell(description: "Name", placeholder: "new contacts nickname")
+    func getText() -> String? {
+		if let text = textField.text {
+			if text.isEmpty {
+				return nil
+			} else {
+				return self.textField.text
+			}
+		} else {
+			return nil
+		}
+	}
+
+
+	func setText(text: String?) {
+		self.textField.text = text
+	}
+	
+
+	static func makeEmailCell(delegate: UITextFieldDelegate? = nil) -> TextFieldCell {
+        let cell = TextFieldCell(description: "Email", placeholder: "you@example.com")
+        cell.textField.keyboardType = .emailAddress
+        // switch off quicktype
+        cell.textField.autocorrectionType = .no
+        cell.textField.autocapitalizationType = .none
+		cell.textField.delegate = delegate
+        return cell
+    }
 
-    nameCell.textField.autocapitalizationType = .words
-    nameCell.textField.autocorrectionType = .no
-    // .namePhonePad doesn't support autocapitalization
-    // see: https://stackoverflow.com/a/36365399
-    // therefore we use .default to capitalize the first character of the name
-    nameCell.textField.keyboardType = .default
+    static func makePasswordCell(delegate: UITextFieldDelegate? = nil) -> TextFieldCell {
+        let cell = TextFieldCell(description: "Password", placeholder: "your IMAP password")
+        cell.textField.textContentType = UITextContentType.password
+        cell.textField.isSecureTextEntry = true
+		return cell
+    }
 
-    return nameCell
-  }
+    static func makeNameCell(delegate: UITextFieldDelegate? = nil) -> TextFieldCell {
+        let cell = TextFieldCell(description: "Name", placeholder: "new contacts nickname")
 
-  static func makeConfigCell(label: String, placeholder: String) -> TextFieldCell {
-    let nameCell = TextFieldCell(description: label, placeholder: placeholder)
+        cell.textField.autocapitalizationType = .words
```

Comment:
there seems to be a lot of whitespace changes in here, that are not really necessary